pandas/tests/indexes/test_interval.py

```diff
@@ -327,171 +335,298 @@ def test_get_item(self):
+
+        index = IntervalIndex.from_tuples(
+            [(0, 2.5), (1, 3), (2, 4)], closed='left')
+
+        # single queries
+        queries = [-0.5, 0, 0.5, 1, 1.5, 2, 2.5, 3, 3.5, 4, 4.5]
+        expected = [(Int64Index([], dtype='int64'), np.array([0]))
+                    (Int64Index([0], dtype='int64'), np.array([]))
+                    (Int64Index([0], dtype='int64'), np.array([]))
+                    (Int64Index([0, 1], dtype='int64'), np.array([]))
+                    (Int64Index([0, 1], dtype='int64'), np.array([]))
+                    (Int64Index([0, 1, 2], dtype='int64'), np.array([]))
+                    (Int64Index([1, 2], dtype='int64'), np.array([]))
+                    (Int64Index([2], dtype='int64'), np.array([]))
+                    (Int64Index([2], dtype='int64'), np.array([]))
+                    (Int64Index([], dtype='int64'), np.array([0]))
+                    (Int64Index([], dtype='int64'), np.array([0]))]
+
+        for query, expected_result in zip(queries, expected):
+            result = index.get_indexer_non_unique([query])
+            tm.assert_numpy_array_equal(result, expected_result)
+
+        # multiple queries
+        queries = [[1, 2], [1, 2, 3], [1, 2, 3, 4], [1, 2, 3, 4, 2]]
+        expected = [(Int64Index([0, 1, 0, 1, 2], dtype='int64'), np.array([]))
+                    (Int64Index([0, 1, 0, 1, 2, 2],
+                                dtype='int64'), np.array([]))
+                    (Int64Index([0, 1, 0, 1, 2, 2, -1],
+                                dtype='int64'), np.array([3]))
+                    (Int64Index([0, 1, 0, 1, 2, 2, -1, 0, 1, 2],
+                                dtype='int64'), np.array([3]))]
+
+        for query, expected_result in zip(queries, expected):
+            result = index.get_indexer_non_unique(query)
+            tm.assert_numpy_array_equal(result, expected_result)
+
+        # TODO we may also want to test get_indexer for the case when
+        # the intervals are duplicated, decreasing, non-monotonic, etc..
+
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_contains_updated_behavior(self):
+
+        index = IntervalIndex.from_arrays([0, 1], [1, 2], closed='right')
+
+        # __contains__ requires perfect matches to intervals.
+        assert 0 not in index
+        assert 1 not in index
+        assert 2 not in index
+
+        assert Interval(0, 1, closed='right') in index
+        assert Interval(0, 2, closed='right') not in index
+        assert Interval(0, 0.5, closed='right') not in index
+        assert Interval(3, 5, closed='right') not in index
+        assert Interval(-1, 0, closed='left') not in index
+        assert Interval(0, 1, closed='left') not in index
+        assert Interval(0, 1, closed='both') not in index
+
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_contains_method_updated_behavior(self):
```

Comment:
As I said before, IMO this is not the desired behaviour (therefore I would leave this for a separate PR to discuss)

Reply:
Is there a reason you keep the current `contains` method tests? I already commented multiple times that they are IMO wrong. You don't have to agree with that :-), but then please answer my comment.

Reply:
Sorry for not answering sooner. I think I changed these tests only slightly, because I felt the original way they operated was okay, but the tests could be a little more extensive (I *think* that's what I was thinking). If you think it should work differently, I'm very open to that, and encourage you to change the method in this PR